test/unit/org/apache/cassandra/index/sai/cql/DropIndexWhileQueryingTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.cql;
+
+import org.apache.cassandra.index.sai.SAITester;
+import org.apache.cassandra.index.sai.plan.QueryController;
+import org.apache.cassandra.inject.ActionBuilder;
+import org.apache.cassandra.inject.Injection;
+import org.apache.cassandra.inject.Injections;
+
+import org.junit.Test;
+
+import static org.apache.cassandra.inject.InvokePointBuilder.newInvokePoint;
+import static org.junit.Assert.assertTrue;
+
+public class DropIndexWhileQueryingTest extends SAITester
+{
+    @Test
+    public void testReproductionForCNDB10732() throws Throwable
```

Comment:
Nit: we might prefer using generic use case names, linking the private issue with a comment, as in:
```
// See CNDB-10732
@Test
public void testDropIndexWhileQuerying() throws Throwable

// See CNDB-10535
@Test
public void testDropVectorIndexWhileQuerying() throws Throwable
```

Reply:
Yeah, I got those for clarity while I was trying to reproduce what is what and forgot to change them, thanks! Totally agree with you!